e2e/framework/provisioning/runner_ssh.go

```diff
@@ -0,0 +1,246 @@
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+// SSHRunner is a ProvisioningRunner that deploys via ssh.
+// Terraform does all of this more elegantly and portably in its
+// ssh communicator, but by shelling out we avoid pulling in TF's as
+// a Nomad dependency, and avoid some long-standing issues with
+// connections to Windows servers. The tradeoff is losing portability
+// but in practice we're always going to run this from a Unixish
+// machine.
+type SSHRunner struct {
+	Key  string // `json:"key"`
+	User string // `json:"user"`
+	Host string // `json:"host"`
+	Port int    // `json:"port"`
+
+	// none of these are available at time of construction, but
+	// should be populated in Open().
+	t               *testing.T
+	controlSockPath string
+	ctx             context.Context
+	cancelFunc      context.CancelFunc
+	copyMethod      func(*SSHRunner, string, string) error
+	muxWait         chan struct{}
+}
+
+// Open establishes the ssh connection. We keep this connection open
+// so that we can multiplex subsequent ssh connections.
+func (runner *SSHRunner) Open(t *testing.T) error {
+	runner.t = t
+	runner.Logf("opening connection to %s", runner.Host)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	runner.ctx = ctx
+	runner.cancelFunc = cancel
+	runner.muxWait = make(chan struct{})
+
+	home, _ := os.UserHomeDir()
+	runner.controlSockPath = filepath.Join(
+		home, ".ssh",
+		fmt.Sprintf("ssh-control-%s-%d.sock", runner.Host, os.Getpid()))
+
+	cmd := exec.CommandContext(ctx,
+		"ssh",
+		"-M", "-S", runner.controlSockPath,
+		"-o", "StrictHostKeyChecking=no", // we're those terrible cloud devs
```

Comment:
lol

Reply:
Managing the whole ssh key setup via [vault](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates/) would be super cool but it'd be a bit of a lift, alas.